simulate/bed.go

```diff
@@ -1,43 +1,71 @@
 package simulate
 
 import (
+	"log"
 	"github.com/vertgenlab/gonomics/bed"
 	"github.com/vertgenlab/gonomics/numbers"
 )
 
+// CountWindows counts the total viable windows of length regionLength in the sequence searchSpace
+func CountWindows(searchSpace []bed.Bed, regionLength int) int {
+	var length, totalWindows int
+	for i := range searchSpace {
+		length = searchSpace[i].ChromEnd - searchSpace[i].ChromStart
+
+		if length >= regionLength {
+			totalWindows += length - regionLength + 1
+		}
+	}
+
+	return totalWindows
+}
+
+// GenerateBedRegion searches the regions of searchSpace (a noGap.bed input file, as a parsed struct) and randomly selects a continuous region of length regionLength 
+func GenerateBedRegion(searchSpace []bed.Bed, randPos int, regionLength int) (bed.Bed, bool) {
```

Comment:
What is randPos?  From the description, it seems like only searchSpace and regionLength would be needed as inputs?

Reply:
I see now that the programmer first needs to needs to get a random number between 0 and CountWindows and pass this as randPos.  I think a description of the allowable range for randPos is all that is needed.

Reply:
Actually I agree with this, I think the randPos should be chosen from either [0, totalWindowNumber-1] or [1, totalWindowNumber]

Reply:
Got it! I added a comment describing randPos to the GeneratebedRegion function. Would you mind quickly reviewing again?

Reply:
I apologize for the delay.  Should it be [0, totalWindows-1]?

Reply:
Yes, thanks for catching!